instantsearch/src/main/java/com/algolia/instantsearch/helpers/Searcher.java

```diff
@@ -164,6 +166,22 @@ public Searcher search(@Nullable final String queryString) {
         return search(queryString, null);
     }
 
+    /**
+     * Starts a search, with the given intent's query if it is a {@link Intent#ACTION_SEARCH search intent}.
+     *
+     * @param intent an Intent that could contain a {@link SearchManager#QUERY QUERY}.
+     * @return this {@link Searcher} for chaining.
+     */
+    @NonNull
+    public Searcher search(@Nullable final Intent intent) {
+        String query = null;
+        if (intent != null && Intent.ACTION_SEARCH.equals(intent.getAction())) {
```

Comment:
supernit: invert to intent.getAction().equals(Intent.ACTION_SEARCH) for readability.

Reply:
Actually can't, cause `getAction` could return `null` :)